src/ngx_http_lua_headers.c

```diff
@@ -145,10 +155,27 @@ ngx_http_lua_ngx_req_raw_header(lua_State *L)
 
     dd("size: %d", (int) size);
 
+#if defined(nginx_version) && nginx_version >= 1011011
+    busy_bufs = ngx_array_create(r->pool, hc->nbusy + 1, sizeof(ngx_buf_t *));
+#endif
+
     if (hc->nbusy) {
         b = NULL;
+
+#if defined(nginx_version) && nginx_version >= 1011011
+        for (cl = hc->busy; cl; /* void */) {
+            bb = ngx_array_push(busy_bufs);
```

Comment:
I think we should check the case that `ngx_array_push` returns NULL. Please also fix other similar places.